bindings/python/BaseTypes.cpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file QuinticSpline.cpp
+ * @authors Diego Ferigo
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <manif/SE3.h>
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "bipedal_locomotion_framework.h"
+
+void BipedalLocomotion::bindings::CreateBaseTypes(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+
+    py::class_<manif::SE3d>(module, "SE3")
+        .def(py::init(
+            []() { return manif::SE3d(Eigen::Vector3d::Zero(), Eigen::Quaterniond::Identity()); }))
+        .def(py::init([](const Eigen::Vector3d& pos, const Eigen::Vector4d& quat) {
+                 return manif::SE3d(pos, quat);
+             }),
+             py::arg("position"),
+             py::arg("quaternion"))
+        .def_property(
+            "position",
+            [](const manif::SE3d& se3) -> Eigen::Vector3d { return se3.coeffs().segment<3>(0); },
+            [](manif::SE3d& se3, const Eigen::Vector3d& position) {
+                se3 = manif::SE3d({position, se3.coeffs().segment<4>(3)});
+            })
+        .def_property(
+            "quaternion",
+            [](const manif::SE3d& se3) -> Eigen::Vector4d { return se3.coeffs().segment<4>(3); },
```

Comment:
Remember that `manif` stores the quaternion like eigen. 
![image](https://user-images.githubusercontent.com/16744101/99458856-7fd71000-292d-11eb-8024-96b5fc115ac4.png)


https://eigen.tuxfamily.org/dox/classEigen_1_1Quaternion.html#a4e6800985f1205f94c2e83260ebe90cf

Reply:
When I inspected `manif` code, it takes the quaternion and position and serialized them to the `coeffs` vector. So, when we extract the quaternion, I take the raw data.

Pybind11 does not know the type `Eigen::Quaternion` so we have in any case to map to to a `Eigen::Vector4d` if we want automatic conversion to a numpy type.